src/core/server/logging/appenders/rolling_file/strategies/numeric/rolling_tasks.ts

```diff
@@ -0,0 +1,96 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { join } from 'path';
+import { getFileNameMatcher, getRollingFileName } from './pattern_matcher';
+import { readdir, rename, unlink, exists } from '../fs';
+
+export const shouldSkipRollout = async ({ logFilePath }: { logFilePath: string }) => {
+  // in case of time-interval triggering policy, we can have an entire
+  // interval without any log event. In that case, the log file is not even
+  // present, and we should not perform the rollout
+  return !(await exists(logFilePath));
+};
+
+/**
+ * Returns the rolled file basenames, from the most recent to the oldest.
+ */
+export const getOrderedRolledFiles = async ({
+  logFileBaseName,
+  logFileFolder,
+  pattern,
+}: {
+  logFileFolder: string;
+  logFileBaseName: string;
+  pattern: string;
+}): Promise<string[]> => {
+  const matcher = getFileNameMatcher(logFileBaseName, pattern);
+  const dirContent = await readdir(logFileFolder);
+  return dirContent
+    .map((fileName) => ({
+      fileName,
+      index: matcher(fileName),
+    }))
+    .filter(({ index }) => index !== undefined)
+    .sort((a, b) => a.index! - b.index!)
+    .map(({ fileName }) => fileName);
+};
+
+export const deleteFiles = async ({
+  logFileFolder,
+  filesToDelete,
+}: {
+  logFileFolder: string;
+  filesToDelete: string[];
+}) => {
+  for (const fileToDelete of filesToDelete) {
+    await unlink(join(logFileFolder, fileToDelete));
```

Comment:
Do we need any exception handling/logging for this?

Reply:
Some tasks are tricky to resume in case of atomic errors. For instance when rolling files in order, if there is an error when renaming a file, we can't really roll the next one as it would throw because the rename target still exists. This is why I'm globally catching all errors from the rolling strategy directly in the RollingFileHandler instead. Do you think this could be improved?

Now for logging the error, I overall agree, see https://github.com/elastic/kibana/pull/84735#discussion_r535490305

Reply:
I agree that resumption is tricky/impossible in some of these scenarios. I can't think of a way to improve what we have, as long as the errors that we're logging are descriptive enough by the time they reach the `RollingFileHandler`. Filesystem errors are usually descriptive enough(?), so I think what you have is fine.

I was asking a legitimate question here, I wasn't necessarily flagging this as an issue